test/functional/pt_osc_migration_functional_test.rb

```diff
@@ -38,6 +38,22 @@ class TestMigration < ActiveRecord::PtOscMigration; end
           ActiveRecord::Base.connection.execute "DROP TABLE IF EXISTS `#{@table_name}`;"
         end
 
+        context 'a raw SQL execute migration' do
+          setup do
+            TestMigration.class_eval <<-EVAL
+            def up
+              execute 'SELECT 1'
+            end
+            EVAL
+          end
+
+          should 'throw an exception' do
+            assert_raise ActiveRecord::UnsupportedMigrationError do
+              @migration.migrate(:up)
+            end
+          end
```

Comment:
Right — by "non-functioning migration" I mean leaving the DB in a bad state because parts of it fail and parts of it succeed due to different setups for `pt-osc` and direct database connections, not it failing to do anything at all (which, as you say, would be preferable).

Essentially: we want to test that if one of these `execute` commands is used, nothing touches the database at all. So also checking that `execute` is never called on the connection/adapter either.